public/app/features/alerting/unified/utils/rule-id.ts

```diff
@@ -244,3 +247,17 @@ function hashLabelsOrAnnotations(item: Labels | Annotations | undefined): string
 export function ruleIdentifierToRuleSourceName(identifier: RuleIdentifier): string {
   return isGrafanaRuleIdentifier(identifier) ? GRAFANA_RULES_SOURCE_NAME : identifier.ruleSourceName;
 }
+
+// DO NOT USE REACT-ROUTER HOOKS FOR THIS CODE
+// React-router's useLocation/useParams/props.match are broken and don't preserve original param values when parsing location
+// so, they cannot be used to parse name and sourceName path params
+// React-router messes the pathname up resulting in a string that is neither encoded nor decoded
+// Relevant issue: https://github.com/remix-run/history/issues/505#issuecomment-453175833
+// It was probably fixed in React-Router v6
+type PathWithOptionalID = { id?: string };
+export function getRuleIdFromPathname(params: PathWithOptionalID): string | undefined {
+  const { pathname = '' } = locationService.getLocation();
```

Comment:
Now that we're patching the router, do we still need this workaround? Can we go back to using `useParams()`?

Reply:
I need to verify. IIRC `react-router-dom` has the same bug. In this PR we only patch the `history` package so it doesn't break outgoing links, but it shouldn't affect parsing path params

Reply:
It seems that react-router added a fix in [here](https://github.com/remix-run/react-router/pull/9477) , and this fix was included in the [v6.4.3](https://github.com/remix-run/react-router/releases/tag/react-router%406.4.3) release